atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/DbkvsPostgresSweeperTest.java

```diff
@@ -15,13 +15,113 @@
  */
 package com.palantir.atlasdb.keyvalue.dbkvs;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
 import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.SweepResults;
 import com.palantir.atlasdb.keyvalue.dbkvs.impl.ConnectionManagerAwareDbKvs;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.DbKvs;
+import com.palantir.atlasdb.keyvalue.impl.SweepStatsKeyValueService;
+import com.palantir.atlasdb.protos.generated.TableMetadataPersistence;
 import com.palantir.atlasdb.sweep.AbstractSweeperTest;
 
 public class DbkvsPostgresSweeperTest extends AbstractSweeperTest {
     @Override
     protected KeyValueService getKeyValueService() {
         return ConnectionManagerAwareDbKvs.create(DbkvsPostgresTestSuite.getKvsConfig());
     }
+
+    @Test
+    public void wholeTableInOneBatch() {
+        setupTestTable();
+        SweepResults results = sweep(TABLE_NAME, 100, 10, 1000);
+        assertThat(results.getNextStartRow().isPresent()).isFalse();
+        assertThat(results.getCellsExamined()).isEqualTo(9);
+        assertThat(results.getCellsDeleted()).isEqualTo(240);
+    }
+
+    @Test
+    public void smallCellBatches() {
+        setupTestTable();
+        SweepResults results = sweep(TABLE_NAME, 100, 10, 7);
+        assertThat(results.getNextStartRow().isPresent()).isFalse();
+        assertThat(results.getCellsExamined()).isEqualTo(9);
+        assertThat(results.getCellsDeleted()).isEqualTo(240);
+    }
+
+    @Test
+    public void smallRangeOfTsMaxBatch() {
+        setupTestTable();
+        setGetRangeOfTsMaxBatch(5);
+        SweepResults results = sweep(TABLE_NAME, 100, 10, 1000);
+        assertThat(results.getNextStartRow().isPresent()).isFalse();
+        assertThat(results.getCellsExamined()).isEqualTo(9);
+        assertThat(results.getCellsDeleted()).isEqualTo(240);
+        resetGetRangeOfTsMaxBatch();
+    }
+
+    @Test
+    public void bothRangesSmallAndSmallerTimestamp() {
+        setupTestTable();
+        setGetRangeOfTsMaxBatch(7);
+        SweepResults results = sweep(TABLE_NAME, 96, 10, 3);
+        assertThat(results.getNextStartRow().isPresent()).isFalse();
+        assertThat(results.getCellsExamined()).isEqualTo(9);
+        assertThat(results.getCellsDeleted()).isEqualTo(213);
+        resetGetRangeOfTsMaxBatch();
+    }
+
+    @Test
+    public void smallRowBatch() {
+        setupTestTable();
+        setGetRangeOfTsMaxBatch(5);
+        SweepResults results = sweep(TABLE_NAME, 100, 5, 1000);
+        assertThat(results.getNextStartRow().isPresent()).isTrue();
+        assertThat(results.getCellsExamined()).isEqualTo(5);
+        assertThat(results.getCellsDeleted()).isEqualTo(170);
+        resetGetRangeOfTsMaxBatch();
+    }
+
+    private void setGetRangeOfTsMaxBatch(long value) {
```

Comment:
Duplicate code (here until the end of the file). Would be nice to have this code in one place only, e.g. something like `DbKvsTestUtils`.